crates/created-swarm/src/swarm.rs

```diff
@@ -159,29 +159,32 @@ where
                 .cloned()
                 .collect::<Vec<_>>();
             let bootstraps = bootstraps(addrs);
-            let (id, node, m_kp, config, http_port) = create_node(bootstraps, addr.clone());
-            ((id, m_kp, config), node, http_port)
+            let (id, node, m_kp, config) = create_node(bootstraps, addr.clone());
+            ((id, m_kp, config), node)
         })
         .collect::<Vec<_>>();
 
-    let http_ports: Vec<u16> = nodes.iter().map(|(_, _, http_port)| *http_port).collect();
-
     // start all nodes
     let infos = join_all(nodes.into_iter().map(
-        move |((peer_id, management_keypair, config), node, _)| {
+        move |((peer_id, management_keypair, config), node)| {
             let connectivity = node.connectivity.clone();
             let aquamarine_api = node.aquamarine_api.clone();
             async move {
-                let outlet = node.start(peer_id).await.expect("node start");
+                let started_node = node.start(peer_id).await.expect("node start");
 
+                let http = started_node
+                    .http_bind_inlet
+                    .await
+                    .expect("Could not bind http");
                 CreatedSwarm {
                     peer_id,
                     multiaddr: config.listen_on,
                     tmp_dir: config.tmp_dir.unwrap(),
                     management_keypair,
-                    outlet,
+                    exit_outlet: started_node.exit_outlet,
                     connectivity,
                     aquamarine_api,
+                    http_bind_addr: http.listen_addr,
```

Comment:
why different names tho?
```suggestion
                    http_listen_addr: http.listen_addr,
```